src/App.tsx

```diff
@@ -187,40 +184,67 @@ function Frame ({
 
   useEffect(() => {
     if (!ready) return
-    if (!frameData ||
-      frameData.season !== parseInt(season || '', 10) ||
-      frameData.episode !== parseInt(episode || '', 10) ||
-      frameData.id !== parseInt(id || '', 10)) {
+    if (!currFrame ||
+      currFrame.season !== parseInt(season || '', 10) ||
+      currFrame.episode !== parseInt(episode || '', 10) ||
+      currFrame.id !== parseInt(id || '', 10)) {
       workerInstance?.loadFrame(season, episode, id).then((data: SearchResult) => {
-        setFrameData(data)
+        setCurrFrame(data)
         setCaption(striptags(data.html))
       })
+      workerInstance?.loadFrame(season, episode, id, -1).then(setPrevFrame)
+      workerInstance?.loadFrame(season, episode, id, 1).then(setNextFrame)
     }
-  }, [frameData, ready, workerInstance, season, episode, id])
+  }, [currFrame, ready, workerInstance, season, episode, id])
 
-  if (!frameData) {
+  if (!currFrame) {
     return <>Loading...</>
   }
   return <div id="selectedItem">
     <Link to={{ pathname: '/', search: searchCriteria && new URLSearchParams({ s: searchCriteria }).toString() }} className="button back">Back to search</Link>
     <canvas ref={canvasRef}></canvas>
     <img ref={imageRef} alt="" />
     <div id="frameNavigation">
-      <button onClick={() => goToRelativeFrame(-1)}>Previous</button>
-      <button onClick={() => goToRelativeFrame(1)}>Next</button>
+      {prevFrame && <Link to={framePath(prevFrame)} className="button" rel="prev">Previous</Link>}
+      {nextFrame && <Link to={framePath(nextFrame)} className="button" rel="next">Next</Link>}
     </div>
     <p>
-      Season {frameData.season} / {' '}
-      Episode {frameData.episode}{' '}
-      ({new Date(frameData.id).toISOString().substr(11, 8)})
+      Season {currFrame.season} / {' '}
+      Episode {currFrame.episode}{' '}
+      ({new Date(currFrame.id).toISOString().substr(11, 8)})
     </p>
     <textarea value={caption} onChange={(event) => setCaption(event.target.value)} aria-label="Caption"></textarea>
     <button onClick={addCurrentFrameToMosaic}>Add to mosaic</button>
     <button onClick={clearMosaic}>Clear mosaic</button>
     <img src={mosaicData} alt="" />
+    {prevFrame && <link rel="preload" href={imageUrl(prevFrame)} as="image" crossOrigin="" />}
```

Comment:
Note: these links [should probably be `rel=prefetch`](https://stackoverflow.com/a/60295034/581845), as that would better match the semantics of what we want here, i.e, prefetching these resources for subsequent navigations, with low priority. And this seems to work perfectly fine on Chrome. But unfortunately on my local testing i couldn't get Firefox to behave as we want: it triggers the prefetching requests, but those prefetched images don't seem to be used when they are needed when navigating to the next page. Instead, Firefox just re-requests them again :(

With `rel=preload` the user-visible behavior is what we want on both browses, but they do log a warning on the console saying `The resource at “https://acrossoverepisode-assets2.storage.googleapis.com/1x04/660911_still.webp” preloaded with link preload was not used within a few seconds. Make sure all attributes of the preload tag are set correctly.`

It's a tradeoff: either we use `rel=preload` and have thing work but not be semantically very correct and have an ugly warning on the console, or we use `rel=prefetch` and things work nice on Chrome but not on Firefox.

Or maybe i just screwed something up and there's a way to make `rel=prefetch` work fine on both browsers hehe